#
# Submitted-By: Erwin Rol, 2006-03-24
# Committed-By: Erwin Rol
#
# Error:
# The bash_launcher plugin sets argv[0] to a fancy name
# this breaks busybox that expects the command name 
# /bin/sh in this case
#
# Description:
#
# State:
# Unknown
#
--- initng-0.6.0RC2/plugins/bash_launcher/initng_bash_launcher.c.orig	2006-03-23 11:43:20.000000000 +0100
+++ initng-0.6.0RC2/plugins/bash_launcher/initng_bash_launcher.c	2006-03-23 11:48:16.000000000 +0100
@@ -99,10 +99,14 @@
 
     /* allocate, and fill the temp argv */
     argtmp = (char **) i_calloc(8, sizeof(char *));
+#if 0
     argtmp[0] = (char *) i_calloc(1, sizeof(char) * (15 + strlen(s->name)));
     strcpy(argtmp[0], "bash_helper[");
     strcat(argtmp[0], s->name);
     strcat(argtmp[0], "]");
+#else
+    argtmp[0] = i_strdup("/bin/sh");
+#endif
     argtmp[1] = i_strdup("-c");
     argtmp[2] = i_strdup(bash_code);
     /* Why do we add service name as argument to shell?!?
